staging/src/k8s.io/component-helpers/apps/podfailureconditions/helpers.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podfailureconditions
```

Comment:
Makes sense to simplify indeed. I have dropped the use of extact from all components except PodGC. This is because these components want to only add or update the condition. 

However, in case of PodGC it is trickier as it also may want to update the phase and not always adds the condition.

If it doesn't extract, it could be that it owns the field already, then by seeing that the pod is already in the terminal phase it would skip the setting of the phase to Failed. As a consequence the phase would be cleared (actually discovered by the integration tests).

This issue could also be solved by unconditionally setting phase in the PodGC - no matter if already in terminal phase or not, but it may produce unnecessary work in case the pod is already in terminal phase. Also, it could unnecessarily move the ownership of the phase to PodGC from another component even if no actual change is needed.